tensorflow/python/keras/engine/network.py

```diff
@@ -1357,10 +1357,40 @@ def load_weights(self, filepath, by_name=False):
       if 'layer_names' not in f.attrs and 'model_weights' in f:
         f = f['model_weights']
       if by_name:
-        saving.load_weights_from_hdf5_group_by_name(f, self.layers)
+        saving.load_weights_from_hdf5_group_by_name(f, self._get_all_layers(self))
       else:
         saving.load_weights_from_hdf5_group(f, self.layers)
 
+  def _get_all_layers (self, network):
```

Comment:
Can you add tests for sequential & functional & subclass models?

Reply:
Will do ti